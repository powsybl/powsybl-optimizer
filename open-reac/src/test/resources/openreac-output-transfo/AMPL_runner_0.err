##### This license is only intended for use by Artelys. #####
##### License is valid until Jan 31, 2024 #####
##### This license is only intended for use by Artelys. #####
##### License is valid until Jan 31, 2024 #####
##### This license is only intended for use by Artelys. #####
##### License is valid until Jan 31, 2024 #####
##### This license is only intended for use by Artelys. #####
##### License is valid until Jan 31, 2024 #####
##### This license is only intended for use by Artelys. #####
##### License is valid until Jan 31, 2024 #####
##### This license is only intended for use by Artelys. #####
##### License is valid until Jan 31, 2024 #####
##### This license is only intended for use by Artelys. #####
##### License is valid until Jan 31, 2024 #####
##### This license is only intended for use by Artelys. #####
##### License is valid until Jan 31, 2024 #####
