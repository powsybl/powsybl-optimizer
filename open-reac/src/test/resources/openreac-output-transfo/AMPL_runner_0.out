



















*** Start of file reactiveopf.run : Wed Jun 21 17:42:59 2023

*** Reading algorithmic controls and parameters in file
PARAM_ALGO_VALUES [*] :=
objective_choice  0
;

Parameter: choice for objective function := 0 (active power minimization)
Parameter: coeff_alpha to choose wether generation vary homogeneously (coeff_alpha=1) or independantly (coeff_alpha=0) is := 1.00
Parameter: threshold to decide wether an active or reactive power value is zero Pnull:=0.01 (MW or Mvar or MVA)
Parameter: threshold to detect zero impedance branch Znull:=0.0001 pu
Parameter: for consistency checks of minimum nominal voltages epsilon_nominal_voltage:= 1 kV
Parameter: for consistency checks of voltage bounds eps<=Vmin<Vmax<=2-eps, epsilon_min_voltage:= 0.5 pu
Parameter: maximum for generating units' parameters Pmin Pmax Qmin Qmax = 9000 MW or Mvar
Parameter: defaultPmax = 1000 MW
Parameter: defaultPmin = 0 MW
Parameter: defaultQmaxPmaxRatio = 0.3 Mvar/MW
Parameter: defaultQmin = -300 Mvar
Parameter: defaultQmax = 300 Mvar
Parameter: minimalQPrange = 1 MW or Mvar

Computation of bus with largest number of branches connected, in order to fix phase to 0 and to choose on which connex component reacive OPF will run
Bus 132kV with most branches: 2 in substation VL_2/VL_2 with 2 connected branches

*** Connexity computation
# CCcomp solve: start (Wed Jun 21 17:42:59 2023)

Artelys Knitro 13.1.0: 
=======================================
             Trial License
       (NOT FOR COMMERCIAL USE)
         Artelys Knitro 13.1.0
=======================================

No start point provided -- Knitro computing one.

Knitro presolve eliminated 2 variables and 3 constraints.

concurrent_evals:        0
datacheck:               0
hessian_no_f:            1
The problem is identified as an LP.
The problem is identified as bound constrained only.
Knitro fixing 2 variables eliminated from the presolve.

Problem Characteristics                                 (   Presolved)
-----------------------
Objective goal:  Maximize
Objective type:  linear
Number of variables:                                  3 (           1)
    bounded below only:                               0 (           0)
    bounded above only:                               0 (           0)
    bounded below and above:                          3 (           0)
    fixed:                                            0 (           1)
    free:                                             0 (           0)
Number of constraints:                                3 (           0)
    linear equalities:                                3 (           0)
    quadratic equalities:                             0 (           0)
    gen. nonlinear equalities:                        0 (           0)
    linear one-sided inequalities:                    0 (           0)
    quadratic one-sided inequalities:                 0 (           0)
    gen. nonlinear one-sided inequalities:            0 (           0)
    linear two-sided inequalities:                    0 (           0)
    quadratic two-sided inequalities:                 0 (           0)
    gen. nonlinear two-sided inequalities:            0 (           0)
Number of nonzeros in Jacobian:                       5 (           0)
Number of nonzeros in Hessian:                        0 (           0)

Knitro using the Interior-Point/Barrier Direct algorithm.

  Iter      Objective      FeasError   OptError    ||Step||    CGits 
--------  --------------  ----------  ----------  ----------  -------
       0    0.000000e+00   0.000e+00

EXIT: Optimal solution found.

Final Statistics
----------------
Final objective value               =   0.00000000000000e+00
Final feasibility error (abs / rel) =   0.00e+00 / 0.00e+00
Final optimality error  (abs / rel) =   9.99e-16 / 9.99e-16
# of iterations                     =          0 
# of CG iterations                  =          0 
# of function evaluations           =          0
# of gradient evaluations           =          0
# of Hessian evaluations            =          0
Total program time (secs)           =       0.001 (     0.000 CPU time)
Time spent in evaluations (secs)    =       0.000

===============================================================================

Knitro 13.1.0: Locally optimal or satisfactory solution.
objective 0; feasibility error 0
0 iterations; 0 function evaluations

suffix feaserror OUT;
suffix opterror OUT;
suffix numfcevals OUT;
suffix numiters OUT;
# CCcomp solve: end   (Wed Jun 21 17:42:59 2023 -> Wed Jun 21 17:42:59 2023)


*** Connexity computation
Nb of busses in AC+DC CC: 3
Nb of busses in CC 'VL_2_0': 3
Nb of busses in other CCs: 0

maximal_voltage_upper_bound = 1.5
minimal_voltage_lower_bound = 0.9
card(SUBSTATIONS) = 3
card(BUS) = 3
card(BUS2) = 3
card(BUSCC) = 3
card(BUS2 diff BUSCC) = 0
card(BUSVV) = 0
card(BUSCC_SLACK) = 2
card(BUSCC diff BUSCC_SLACK) = 1
card(BRANCH) = 2
card(BRANCHCC) = 2
card(BRANCHZNULL) = 0
card(BRANCHCC diff BRANCHZNULL) = 2
card(UNIT) = 1
card(UNITCC) = 1
card(UNITON) = 1
card(UNITON diff UNIT_FIXQ) = 1
card(UNIT_FIXQ) = 0
card(LOAD) = 2
card(LOADCC) = 2
card(SHUNTCC) = 0
card(SHUNT_FIX) = 0
card(SHUNT_VAR) = 0
card(SVC) = 0
card(SVCCC) = 0
card(SVCON) = 0
card(VSCCONV) = 0
card(VSCCONVON) = 0
card(LCCCONV) = 0
card(LCCCONVON) = 0

HVDC injections (homogeneous to loads):
Sum of HVDC conv.  H: 0 MW
Sum of loads       C: 16 MW
Sum of generations P: 25 MW
  (including batteries for 0.0 MW
Balance    (P-C-H)/C: 54.32 %    (global_initial_losses_ratio=0.543210)

Branch 'LINE_12' Fmax=22862.17MW is small ; Vnom1=132kV Vnom2=132kV patl1=-99999A patl2=-99999A (Fmax not used, this is just for information)
Branch 'T2wT' Fmax=22862.17MW is small ; Vnom1=132kV Vnom2=33kV patl1=-99999A patl2=-99999A (Fmax not used, this is just for information)
Substations 'VL_1' with lowest  voltage lower bound Vnom=132kV Vmin=0.900pu
Substations 'VL_2' with lowest  voltage lower bound Vnom=132kV Vmin=0.900pu
Substations 'VL_3' with highest voltage upper bound Vnom=33kV Vmax=3.030pu
If voltage lower bounds are missing or too small, they are set to 0.900pu
If voltage upper bounds are missing or too high,  they are set to 1.500pu
Maximal diameter of voltage interval: 0.600
Minimal diameter of voltage interval: 0.200
Minimal transformer ratio : 0.900
Maximal transformer ratio : 1.100
Transformer 'T2wT' ratio_min=0.900 ratio_max=1.100


Raw extremal values: 0 <= 'unit_Pmin' <= 0
Active generation:   25 <= 'unit_Pc' <= 25
Raw extremal values: 140 <= 'unit_Pmax' <= 140
Raw extremal values: -1.79769e+308 <= 'unit_qP' <= -1.79769e+308
Raw extremal values: -1.79769e+308 <= 'unit_qp' <= -1.79769e+308
Raw extremal values: 1.79769e+308 <= 'unit_QP' <= 1.79769e+308
Raw extremal values: 1.79769e+308 <= 'unit_Qp' <= 1.79769e+308
Corrected extremal values: 0 <= 'corrected_unit_Pmin' <= 0
Corrected extremal values: 140 <= 'corrected_unit_Pmax' <= 140
Corrected extremal values: -42 <= 'corrected_unit_qP' <= -42
Corrected extremal values: -42 <= 'corrected_unit_qp' <= -42
Corrected extremal values: 42 <= 'corrected_unit_QP' <= 42
Corrected extremal values: 42 <= 'corrected_unit_Qp' <= 42

######################################################################
# DCopf solve: start (Wed Jun 21 17:42:59 2023)

Artelys Knitro 13.1.0: 
=======================================
             Trial License
       (NOT FOR COMMERCIAL USE)
         Artelys Knitro 13.1.0
=======================================

No start point provided -- Knitro computing one.

Knitro presolve eliminated 2 variables and 2 constraints.

concurrent_evals:        0
datacheck:               0
hessian_no_f:            1
The problem is identified as a convex QP.
Knitro reinitializing 2 variables eliminated from the presolve.

Problem Characteristics                                 (   Presolved)
-----------------------
Objective goal:  Minimize
Objective type:  quadratic
Number of variables:                                 11 (           9)
    bounded below only:                               6 (           6)
    bounded above only:                               0 (           0)
    bounded below and above:                          2 (           2)
    fixed:                                            0 (           0)
    free:                                             3 (           1)
Number of constraints:                                5 (           3)
    linear equalities:                                5 (           3)
    quadratic equalities:                             0 (           0)
    gen. nonlinear equalities:                        0 (           0)
    linear one-sided inequalities:                    0 (           0)
    quadratic one-sided inequalities:                 0 (           0)
    gen. nonlinear one-sided inequalities:            0 (           0)
    linear two-sided inequalities:                    0 (           0)
    quadratic two-sided inequalities:                 0 (           0)
    gen. nonlinear two-sided inequalities:            0 (           0)
Number of nonzeros in Jacobian:                      15 (          11)
Number of nonzeros in Hessian:                        1 (           1)

Knitro using the Interior-Point/Barrier Direct algorithm.

  Iter      Objective      FeasError   OptError    ||Step||    CGits 
--------  --------------  ----------  ----------  ----------  -------
       0    1.006456e+04   1.124e+01
       4    7.744007e+01   7.848e-08   4.422e-08   5.890e-04        0

EXIT: Optimal solution found.

Final Statistics
----------------
Final objective value               =   7.74400713111799e+01
Final feasibility error (abs / rel) =   7.85e-08 / 6.98e-09
Final optimality error  (abs / rel) =   4.42e-08 / 4.42e-10
# of iterations                     =          4 
# of CG iterations                  =          0 
# of function evaluations           =          0
# of gradient evaluations           =          0
# of Hessian evaluations            =          0
Total program time (secs)           =       0.002 (     0.000 CPU time)
Time spent in evaluations (secs)    =       0.000

===============================================================================

Knitro 13.1.0: Locally optimal or satisfactory solution.
objective 77.44007131; feasibility error 7.85e-08
4 iterations; 0 function evaluations

# DCopf solve: end   (Wed Jun 21 17:42:59 2023 -> Wed Jun 21 17:42:59 2023)
######################################################################

OK all slack variables for DCOPF are null
Maximum flow: 16.20 MW
Maximum flow 16.20 MW is on branch 'LINE_12'
Maximum deviation between generation and target: 8.80 MW
Sum of HVDC conv.  H: 0 MW
Sum of loads       C: 16 MW
Sum of generations P: 16 MW
Balance    (P-C-H)/C: -0.00 %

teta_max = 3.0093
teta_min = -3.04591
max({n in BUSCC} (teta_dc[n])) = 0.00929752
min({n in BUSCC} (teta_dc[n])) = -0.0459137
max({(qq,m,n) in BRANCHCC} (teta_dc[m] - teta_dc[n])) = 0.0459137
min({(qq,m,n) in BRANCHCC} (teta_dc[m] - teta_dc[n])) = 0.00929752

Branches with large Delta Teta:
'T2wT' R=1.5611 X=0.9183 delta=0.05 teta_m=0.00 teta_n=-0.05 F=5.00

######################################################################
# ACopf solve: start (Wed Jun 21 17:42:59 2023)

Artelys Knitro 13.1.0: opttol=1
opttolabs=1e-1
feastol=1
feastolabs=1e-3
maxit=200

=======================================
             Trial License
       (NOT FOR COMMERCIAL USE)
         Artelys Knitro 13.1.0
=======================================

Knitro presolve eliminated 0 variables and 0 constraints.

concurrent_evals:        0
datacheck:               0
feastol:                 1
hessian_no_f:            1
maxit:                   200
opttol:                  1
opttol_abs:              0.1
Knitro shifted start point further inside presolved bounds (4 variables).

Problem Characteristics                                 (   Presolved)
-----------------------
Objective goal:  Minimize
Objective type:  quadratic
Number of variables:                                 12 (          12)
    bounded below only:                               0 (           0)
    bounded above only:                               0 (           0)
    bounded below and above:                         12 (          12)
    fixed:                                            0 (           0)
    free:                                             0 (           0)
Number of constraints:                                6 (           6)
    linear equalities:                                0 (           0)
    quadratic equalities:                             0 (           0)
    gen. nonlinear equalities:                        6 (           6)
    linear one-sided inequalities:                    0 (           0)
    quadratic one-sided inequalities:                 0 (           0)
    gen. nonlinear one-sided inequalities:            0 (           0)
    linear two-sided inequalities:                    0 (           0)
    quadratic two-sided inequalities:                 0 (           0)
    gen. nonlinear two-sided inequalities:            0 (           0)
Number of nonzeros in Jacobian:                      32 (          32)
Number of nonzeros in Hessian:                       16 (          16)

Knitro using the Interior-Point/Barrier Direct algorithm.

  Iter      Objective      FeasError   OptError    ||Step||    CGits 
--------  --------------  ----------  ----------  ----------  -------
       0    6.580514e+01   2.064e+01
       6    2.286872e+01   2.315e-05   1.120e-05   1.385e-03        0

EXIT: Locally optimal solution found.

Final Statistics
----------------
Final objective value               =   2.28687178882257e+01
Final feasibility error (abs / rel) =   2.32e-05 / 1.12e-06
Final optimality error  (abs / rel) =   1.12e-05 / 1.12e-07
# of iterations                     =          6 
# of CG iterations                  =          0 
# of function evaluations           =          8
# of gradient evaluations           =          8
# of Hessian evaluations            =          6
Total program time (secs)           =       0.005 (     0.000 CPU time)
Time spent in evaluations (secs)    =       0.002

===============================================================================

Knitro 13.1.0: Locally optimal or satisfactory solution.
objective 22.86871789; feasibility error 2.32e-05
6 iterations; 8 function evaluations

# ACopf solve: end   (Wed Jun 21 17:42:59 2023 -> Wed Jun 21 17:42:59 2023)
######################################################################

nb_iter_last = 6
nb_iter_total = 6
max({(qq,m,n) in BRANCHCC} (branch_R[1,qq,m,n])) = 1.56107
max({(qq,m,n) in BRANCHCC} (branch_X[1,qq,m,n])) = 0.918274
teta_max = 3.0093
max({n in BUSCC} (teta[n])) = 0.0128223
max({n in BUSCC} (teta_dc[n])) = 0.00929752
teta_min = -3.04591
min({n in BUSCC} (teta[n])) = -0.0516555
min({n in BUSCC} (teta_dc[n])) = -0.0459137
max({(qq,m,n) in BRANCHCC} (teta[m] - teta[n])) = 0.0516555
max({(qq,m,n) in BRANCHCC} (teta_dc[m] - teta_dc[n])) = 0.0459137
min({(qq,m,n) in BRANCHCC} (teta[m] - teta[n])) = 0.0128223
min({(qq,m,n) in BRANCHCC} (teta_dc[m] - teta_dc[n])) = 0.00929752
min({n in BUSCC} (V[n])) = 0.9
max({n in BUSCC} (V[n])) = 0.995834

'T2wT' R=1.561 X=0.918 delta=0.05 Vm=0.990 0.00 Vn=0.900 -0.05 sQm 0.00 0.00 sQn 0.00 0.00 P1=11.6 Q1=-0.3 P2=-5.5 Q2=0.0 

*** End of file reactiveopf.run : Optimization successfull
*** Start of file reactiveopf.run : 'Wed Jun 21 17:42:59 2023'
*** End   of file reactiveopf.run : 'Wed Jun 21 17:42:59 2023'
